### ISSUE TITLE
Formula for calculating maxHfov from voav and screen ratio for partial panoramas

### ISSUE THREAD

**Issue by DStillingfleet** [mpetroff/pannellum#521]
Hi

I'm looking for a formula to calculate the hfov and maxHfov of a panorama from the voav and screen ratio for a partial panorama. Any ideas?

Here is the scenario.

I have some 360 degree panoramas, that vary in their voav (height), for example 67, to 62 degrees. 

The maxPitch and MinPitch values are easily calculated and work very effectively when the hfov and maxHfov are also set appropriately, i.e. there is no black band at the top and bottom of the panorama where the view exceeds the voav (height) of the image.

The ideal hfov and maxHfov vary with the display ratio.  I.e. the narrower the display width, the narrower the hfov.  But it's not linear, and varies with voav.

Many thanks in advance

Derek

**Comment by mpetroff** [comment]
Here's the line from `libpannellum.js` that calculates the vertical field of view from the horizontal field of view: `var vfov = 2 * Math.atan(Math.tan(hfov * 0.5) / (canvas.clientWidth / canvas.clientHeight));`. Hopefully, this points you in the right direction.

**Comment by DStillingfleet** [comment]
Hi Matthew

Many thanks for the hint.  

I've inversed and adapted your formula to .....

"maxHfov=((Math.atan(ScreenRatio*(Math.tan((Voav/100)/1.15))))*115);"

Not sure why the dividing by 1.15 gives the correct results rather than 2 which gave too high results on wider windows. Found by trial and error.

This moves me onto the next stage where I need to reset the maxHfov after the window is resized. 

I anticipated that the "setHfovBounds(min,max)" would do the trick.  But it does not appear to be working, or I'm using it incorrectly.  With "getHfovBounds" I can read the existing values before the "setHfovBounds" function.  But when I try the "getHfovBounds" after using the "setHfovBounds" it returns "NaN,NaN".

Once this is resolved, I'm expecting to be able to call the "resize" to resize the partial panorama so that no black is showing at the top and bottom.  Would that be correct.

Kind regards


Derek
